addons/web/static/src/js/views/form_widgets.js

```diff
@@ -383,87 +384,111 @@ var FieldFloat = FieldChar.extend({
 
-        if (this.get('value')) {
-            var model = this.options.model || this.field_manager.get_field_value(this.options.model_field);
-            try{
-                var domain = pyeval.eval('domain', this.get('value'));
-            }
-            catch(e){
-                this.do_warn(_t('Error: Bad domain'), _t('The domain is wrong.'));
-                return;
-            }
-            var ds = new data.DataSetStatic(self, model, self.build_context());
-            ds.call('search_count', [domain]).then(function (results) {
-                self.$('.o_count').text(results + _t(' selected records'));
-                if (self.get('effective_readonly')) {
-                    self.$('button').text(_t('See selection '));
-                }
-                else {
-                    self.$('button').text(_t('Change selection '));
-                }
-                self.$('button').append($("<span/>").addClass('fa fa-arrow-right'));
-            });
+        var domain = pyeval.eval("domain", this.get("value") || "[]");
 
-            if(this.debug) {
-                this.$('.o_debug_input').val(this.get('value'));
+        // Recreate domain widget with new domain value
+        if (!this._noDomainSelectorRender) {
+            if (this.domainSelector) {
+                this.domainSelector.destroy();
             }
-        } else {
-            this.$('.o_form_input').val('');
-            this.$('.o_count').text(_t('No selected record'));
-            var $arrow = this.$('button span').detach();
-            this.$('button').text(_t('Select records ')).append($("<span/>").addClass('fa fa-arrow-right'));
+            this.domainSelector = new DomainSelector(this, this.model, domain, {
+                readonly: this.get("effective_readonly"),
+                fs_filters: this.options.fs_filters,
+            });
+            this.domainSelector.prependTo(this.$el);
         }
-    },
-    on_click: function(event) {
-        event.preventDefault();
 
-        var self = this;
-        var dialog = new common.DomainEditorDialog(this, {
-            res_model: this.options.model || this.field_manager.get_field_value(this.options.model_field),
-            default_domain: this.get('value'),
-            title: this.get('effective_readonly') ? _t('Selected records') : _t('Select records...'),
-            readonly: this.get('effective_readonly'),
-            disable_multiple_selection: this.get('effective_readonly'),
-            no_create: this.get('effective_readonly'),
-            on_selected: function(selected_ids) {
-                if (!self.get('effective_readonly')) {
-                    self.set_value(dialog.get_domain(selected_ids));
-                }
-            }
+        // Show number of selected records
+        new Model(this.model).call("search_count", [domain], {
+            context: this.build_context(),
+        }).then((function (data) {
+            this.valid = true;
```

Comment:
Don't like it either but today, I prefer bind(this) to self (we will see tomorrow if that changes :))